oneandone/src/test/java/org/apache/jclouds/oneandone/rest/internal/BaseOneAndOneLiveTest.java

```diff
@@ -0,0 +1,100 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.oneandone.rest.internal;
+
+import com.google.common.base.Predicate;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import org.apache.jclouds.oneandone.rest.OneAndOneApi;
+import org.apache.jclouds.oneandone.rest.OneAndOneProviderMetadata;
+import org.apache.jclouds.oneandone.rest.config.OneAndOneConstants;
+import org.apache.jclouds.oneandone.rest.config.OneAndOneProperties;
+import org.apache.jclouds.oneandone.rest.domain.Hardware;
+import org.apache.jclouds.oneandone.rest.domain.Hdd;
+import org.apache.jclouds.oneandone.rest.domain.Server;
+import org.apache.jclouds.oneandone.rest.domain.Types;
+import org.jclouds.apis.BaseApiLiveTest;
+import org.jclouds.util.Predicates2;
+import static org.testng.Assert.assertTrue;
+
+public class BaseOneAndOneLiveTest extends BaseApiLiveTest<OneAndOneApi> {
+
+    private Predicate<Server> serverReady;
+    private static final OneAndOneProviderMetadata METADATA = new OneAndOneProviderMetadata();
+    OneAndOneConstants constants;
+
+    public BaseOneAndOneLiveTest() {
+        provider = "oneandone";
+    }
+
+    @Override
+    protected Properties setupProperties() {
+        Properties props = super.setupProperties();
+        setIfTestSystemPropertyPresent(props, OneAndOneProperties.AUTH_TOKEN);
+        return props;
+    }
+
+    @Override
+    protected OneAndOneApi create(Properties props, Iterable<Module> modules) {
+        Injector injector = newBuilder().modules(modules).overrides(props).buildInjector();
+        constants = injector.getInstance(OneAndOneConstants.class);
+        Predicate<Server> serverAvailableCheck = new Predicate<Server>() {
+            @Override
+            public boolean apply(Server currentServer) {
+                Server server = api.serverApi().get(currentServer.id());
+
+                if ((server.status().state() != Types.ServerState.POWERED_OFF
+                        && server.status().state() != Types.ServerState.POWERED_ON)
+                        || server.status().percent() != 0) {
+                    return false;
+                } else {
+                    return true;
+                }
+            }
+        };
```

Comment:
Is there an example for this in the HttpApiModule this is in the ComputeContextModule, maybe leave it here until we get to the compute and then added it to the ContextModule?